services/requester/requester.go

```diff
@@ -310,6 +310,11 @@ func (e *EVM) EstimateGas(
 	height uint64,
 	stateOverrides *ethTypes.StateOverride,
 ) (uint64, error) {
+	iterations := 0
+	defer func() {
+		e.collector.GasEstimationIterations(iterations)
```

Comment:
Are there any concerns about reporting the iterations for the case where this function errors?

Reply:
Good point :+1: I guess we should not report iterations for calls/transactions that we know would error out / or revert. It will skew our metrics, as we know that we need 2+ iterations to estimate the gas, with the currently defined error ratio of `1.5%`.
Updated in https://github.com/onflow/flow-evm-gateway/pull/725/commits/c9cb6120d643213948e55a5b71d94322bf6a9b21#diff-dead54ca0eb128b19b453911c9bc13c64e01b2c50553a574820820564552e39bR350-R355